x/crosschain/types/status.go

```diff
@@ -9,44 +9,35 @@ func (m *Status) AbortRefunded() {
 	m.StatusMessage = "CCTX aborted and Refunded"
 }
 
-// UpdateStatusAndErrorMessages transitions the Status and Error messages.
-func (m *Status) UpdateStatusAndErrorMessages(newStatus CctxStatus, statusMsg, errorMsg string) {
-	m.UpdateStatus(newStatus, statusMsg)
-
-	if errorMsg != "" {
-		m.UpdateErrorMessage(errorMsg)
-	}
+func (m *Status) UpdateStatusAndErrorMessages(newStatus CctxStatus, messages StatusMessages) {
+	m.UpdateStatus(newStatus)
+	m.UpdateErrorMessages(messages)
 }
 
-// UpdateStatus updates the cctx status and cctx.status.status_message.
-func (m *Status) UpdateStatus(newStatus CctxStatus, statusMsg string) {
+// UpdateStatus updates the cctx status
+func (m *Status) UpdateStatus(newStatus CctxStatus) {
 	if m.ValidateTransition(newStatus) {
-		m.StatusMessage = fmt.Sprintf("Status changed from %s to %s", m.Status.String(), newStatus.String())
 		m.Status = newStatus
 	} else {
 		m.StatusMessage = fmt.Sprintf(
 			"Failed to transition status from %s to %s",
 			m.Status.String(),
 			newStatus.String(),
 		)
-
 		m.Status = CctxStatus_Aborted
 	}
-
-	if statusMsg != "" {
-		m.StatusMessage += fmt.Sprintf(": %s", statusMsg)
-	}
 }
 
-// UpdateErrorMessage updates cctx.status.error_message.
-func (m *Status) UpdateErrorMessage(errorMsg string) {
-	errMsg := errorMsg
-
-	if errMsg == "" {
-		errMsg = "unknown error"
+// UpdateErrorMessages updates cctx.status.error_message and cctx.status.error_message_revert.
+func (m *Status) UpdateErrorMessages(messages StatusMessages) {
+	// Always update the status message, status should contain only the most recent update
+	m.StatusMessage = messages.StatusMessage
```

Comment:
should do
```
if messages.ErrorMessageOutbound != "" {
		m.ErrorMessage = messages.ErrorMessageOutbound
	}
```
as well

Reply:
do you mean 

if m.StatusMessage != "" {} 
I decided not to add the check to allow empty status messages if we ever need it ( However, we do not have empty messages now, as every update has some update for status )